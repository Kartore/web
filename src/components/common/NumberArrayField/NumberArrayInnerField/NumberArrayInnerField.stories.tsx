import type { Meta, StoryObj } from '@storybook/react';

import { NumberArrayInnerField } from '.';

const meta = {
  component: NumberArrayInnerField,
} satisfies Meta<typeof NumberArrayInnerField>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    label: 'X',
    value: 0,
  },
};
