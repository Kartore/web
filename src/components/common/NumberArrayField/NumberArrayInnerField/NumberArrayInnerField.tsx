import type { FC } from 'react';
import { useRef } from 'react';

import {
	type AriaNumberFieldProps,
	useLocale,
	useNumberField,
} from 'react-aria';
import { useNumberFieldState } from 'react-stately';

import { cn } from '~/utils/tailwindUtil';

export type NumberArrayInnerFieldProps = {
	className?: string;
} & AriaNumberFieldProps;

export const NumberArrayInnerField: FC<NumberArrayInnerFieldProps> = ({
	className,
	label,
	...props
}) => {
	const { locale } = useLocale();
	const state = useNumberFieldState({ ...props, label, locale });
	const ref = useRef(null);
	const { labelProps, inputProps, groupProps } = useNumberField(
		{ ...props, label },
		state,
		ref,
	);

	return (
		<div
			className={cn(
				'flex flex-row items-center justify-between gap-2 rounded bg-gray-100 px-2 py-0.5',
				className,
			)}
		>
			<label
				{...labelProps}
				className={cn(
					'font-semibold text-gray-500 text-sm',
					labelProps.className,
				)}
			>
				{label}
			</label>
			<div {...groupProps} className={'w-full'}>
				<input
					{...inputProps}
					ref={ref}
					className={cn(
						'w-full rounded border-none px-1 py-0.5 font-semibold text-sm transition-colors hover:bg-gray-200 focus-visible:bg-gray-200 focus-visible:outline-0',
						inputProps.className,
					)}
				/>
			</div>
		</div>
	);
};

NumberArrayInnerField.displayName = 'NumberArrayInnerField';
