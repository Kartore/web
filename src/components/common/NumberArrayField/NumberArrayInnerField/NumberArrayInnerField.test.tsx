import { describe } from 'vitest';
import { render } from '@testing-library/react';
import { composeStories } from '@storybook/react';

import * as Stories from './NumberArrayInnerField.stories';

const { Default } = composeStories(Stories);

describe('Component: NumberArrayInnerField', () => {
  describe('Snapshot', () => {
    it('Default', () => {
      const { asFragment } = render(<Default />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
