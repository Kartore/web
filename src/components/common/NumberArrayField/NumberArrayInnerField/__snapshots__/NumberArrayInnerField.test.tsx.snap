// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component: NumberArrayInnerField > Snapshot > Default 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row items-center justify-between gap-2 rounded bg-gray-100 py-0.5 px-2"
  >
    <label
      class="text-sm font-semibold text-gray-500"
      for="react-aria-:r0:"
      id="react-aria-:r2:"
    >
      X
    </label>
    <div
      class="w-full"
      role="group"
    >
      <input
        aria-labelledby="react-aria-:r2:"
        aria-roledescription="Number field"
        autocomplete="off"
        autocorrect="off"
        class="w-full rounded border-none py-0.5 px-1 text-sm font-semibold transition-colors hover:bg-gray-200 focus-visible:bg-gray-200 focus-visible:outline-0"
        id="react-aria-:r0:"
        inputmode="numeric"
        spellcheck="false"
        type="text"
        value="0"
      />
    </div>
  </div>
</DocumentFragment>
`;
